pkg/pillar/zedbox/zedbox.go

```diff
@@ -78,31 +78,31 @@ var (
 		"client":           {f: client.Run, inline: inlineAlways},
 		"command":          {f: command.Run},
 		"diag":             {f: diag.Run, inline: inlineUnlessService},
-		"domainmgr":        {f: domainmgr.Run},
-		"downloader":       {f: downloader.Run},
+		"domainmgr":        {f: domainmgr.Run, inline: inlineUnlessService},
+		"downloader":       {f: downloader.Run, inline: inlineUnlessService},
 		"executor":         {f: executor.Run},
 		"faultinjection":   {f: faultinjection.Run},
 		"hardwaremodel":    {f: hardwaremodel.Run, inline: inlineAlways},
-		"ledmanager":       {f: ledmanager.Run},
-		"loguploader":      {f: loguploader.Run},
-		"nim":              {f: nim.Run},
-		"nodeagent":        {f: nodeagent.Run},
-		"verifier":         {f: verifier.Run},
-		"volumemgr":        {f: volumemgr.Run},
+		"ledmanager":       {f: ledmanager.Run, inline: inlineUnlessService},
+		"loguploader":      {f: loguploader.Run, inline: inlineUnlessService},
+		"nim":              {f: nim.Run, inline: inlineUnlessService},
+		"nodeagent":        {f: nodeagent.Run, inline: inlineUnlessService},
+		"verifier":         {f: verifier.Run, inline: inlineUnlessService},
+		"volumemgr":        {f: volumemgr.Run, inline: inlineUnlessService},
 		"waitforaddr":      {f: waitforaddr.Run, inline: inlineAlways},
-		"zedagent":         {f: zedagent.Run},
-		"zedmanager":       {f: zedmanager.Run},
-		"zedrouter":        {f: zedrouter.Run},
+		"zedagent":         {f: zedagent.Run, inline: inlineUnlessService},
+		"zedmanager":       {f: zedmanager.Run, inline: inlineUnlessService},
+		"zedrouter":        {f: zedrouter.Run, inline: inlineUnlessService},
 		"ipcmonitor":       {f: ipcmonitor.Run, inline: inlineAlways},
-		"baseosmgr":        {f: baseosmgr.Run},
-		"wstunnelclient":   {f: wstunnelclient.Run},
+		"baseosmgr":        {f: baseosmgr.Run, inline: inlineUnlessService},
+		"wstunnelclient":   {f: wstunnelclient.Run, inline: inlineUnlessService},
 		"conntrack":        {f: conntrack.Run, inline: inlineAlways},
 		"pbuf":             {f: pbuf.Run, inline: inlineAlways},
 		"tpmmgr":           {f: tpmmgr.Run, inline: inlineUnlessService},
 		"vaultmgr":         {f: vaultmgr.Run, inline: inlineUnlessService},
```

Comment:
A separate PR to introduce a inlineIfNoArgs to avoid the runAsService argument to tpmmgr and vaultmgr might be a good preparation to make those be more regular.

Reply:
Good idea, opened: https://github.com/lf-edge/eve/pull/2845